src/python/examples/unittests/test_data_dumps.py

```diff
@@ -0,0 +1,193 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `data_dumps.py` script.
+
+Typical usage example::
+
+    $ pytest test_data_dumps.py
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+from importlib.abc import Loader
+from importlib.machinery import ModuleSpec
+from importlib.util import module_from_spec, spec_from_file_location
+import os
+from pathlib import Path
+import sys
+from typing import ContextManager, Dict, List
+
+import sqlalchemy
+
+import pytest
+
+from ensembl.compara.filesys import file_cmp
+
+
+script_path = Path(__file__).parents[3] / "scripts" / "pipeline" / "data_dumps.py"
+script_name = script_path.stem
+script_spec = spec_from_file_location(script_name, script_path)
+
+if not isinstance(script_spec, ModuleSpec):
+    raise ImportError(f"ModuleSpec not created for module file '{script_path}'")
+if not isinstance(script_spec.loader, Loader):
+    raise ImportError(f"no loader found for module file '{script_path}'")
+
+data_dumps_module = module_from_spec(script_spec)
+sys.modules[script_name] = data_dumps_module
+script_spec.loader.exec_module(data_dumps_module)
+
+# pylint: disable=import-error,wrong-import-order,wrong-import-position
+
+import data_dumps  # type: ignore
+
+# pylint: enable=import-error,wrong-import-order,wrong-import-position
+
+def test_find_latest_core_naive() -> None:
+    """Tests :func:`data_dumps.find_latest_core()` function when it does not raise an error.
+
+    Args:
+        core_names: A list of core database names.
+
+    """
+    assert data_dumps.find_latest_core(
+        ["mus_musculus_core_105_1", "mus_musculus_core_52_105_3", "mus_musculus_core_104_4"]
+            ) == "mus_musculus_core_52_105_3"
```

Comment:
I think assigning to a `core_names` variable makes it easier to compare this to the parametrised version of the test:
```suggestion
    core_names = ["mus_musculus_core_105_1", "mus_musculus_core_52_105_3", "mus_musculus_core_104_4"]
    assert data_dumps.find_latest_core(core_names) == "mus_musculus_core_52_105_3"
```